File: src/agent/test-fixtures/reply-event-script.mjs
let raw = "";
process.stdin.setEncoding("utf8");
process.stdin.on("data", (chunk) => {
  raw += chunk;
});
process.stdin.on("end", () => {
  const payload = JSON.parse(raw || "{}");
  const eventType = payload?.message?.eventType ?? "unknown";
  const eventKey = payload?.message?.eventKey ?? "";
  const changeType = payload?.message?.changeType ?? "";

  if (eventKey === "PASS_TO_DEFAULT") {
    process.stdout.write(JSON.stringify({
      ok: true,
      action: "none",
      chainToAgent: true,
    }));
    return;
  }

  process.stdout.write(JSON.stringify({
    ok: true,
    action: "reply_text",
    reply: {
      text: `script:${eventType}:${eventKey}:${changeType}`,
    },
    chainToAgent: false,
  }));
});

File: src/agent/test-fixtures/reply-event-script.py
import json
import sys

payload = json.load(sys.stdin)
message = payload.get("message", {})
event_type = message.get("eventType", "unknown")
event_key = message.get("eventKey") or ""
change_type = message.get("changeType") or ""

json.dump({
    "ok": True,
    "action": "reply_text",
    "reply": {
        "text": f"python:{event_type}:{event_key}:{change_type}"
    },
    "chainToAgent": False
}, sys.stdout)
